google-cloud-testing/spanner-benchwrapper/src/main/java/com/google/cloud/benchwrapper/SpannerBenchWrapperImpl.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.benchwrapper;
+
+import static com.google.cloud.spanner.TransactionRunner.TransactionCallable;
+
+import com.google.api.core.ApiFunction;
+import com.google.cloud.benchwrapper.SpannerBenchWrapperGrpc.SpannerBenchWrapperImplBase;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ReadOnlyTransaction;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerBatchUpdateException;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import java.util.ArrayList;
+import java.util.List;
+
+class SpannerBenchWrapperImpl extends SpannerBenchWrapperImplBase {
+  private Spanner spanner;
+  private DatabaseClient dbClient;
+
+  public SpannerBenchWrapperImpl(String spannerEmulatorHost) {
+    SpannerOptions.Builder builder =
+        SpannerOptions.newBuilder()
+            .setHost("http://" + spannerEmulatorHost)
```

Comment:
Given that we generically want the Java client lib to support an emulator so that users can use an emulator without having to change the code on how they init the client lib, I think it would be better if we do this in the client lib code.

This is how it's done in [Go](https://github.com/googleapis/google-cloud-go/blob/1118b0760db3b1989320683fd9ce284a48749a6a/spanner/client.go#L132) currently and the [Python PR](https://github.com/googleapis/google-cloud-python/pull/9902) is the same.